lib/srv/regular/sshserver_test.go

```diff
@@ -1411,13 +1411,51 @@ func TestClientDisconnect(t *testing.T) {
 	require.NoError(t, clt.Close())
 }
 
+func testHelperWaitForServerStart(t *testing.T, s *Server, config *ssh.ClientConfig, timeout time.Duration) {
+	for end :=  time.After(timeout); true;{
+		select {
+		case <-end:
+			t.Fatalf("Server failed to start after: %v", timeout)
+			return
+		default:
+		}
+		clt, _ := ssh.Dial("tcp", s.Addr(), config)
+		if clt != nil{
+			require.NoError(t, clt.Close())
+			require.ErrorIs(t, clt.Wait(), net.ErrClosed)
+			return
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+type fakeClock struct {
+	clock clockwork.FakeClock
+}
+
+func (fc *fakeClock) UtcNow() time.Time {
+	return fc.clock.Now()
+}
+
+func (fc *fakeClock) Sleep(d time.Duration) {
+	fc.clock.Sleep(d)
+}
+
+func (fc *fakeClock) After(d time.Duration) <-chan time.Time {
+	return fc.clock.After(d)
+}
+
 func TestLimiter(t *testing.T) {
 	t.Parallel()
 	f := newFixture(t)
 	ctx := context.Background()
 
+	fClock := &fakeClock{
+		f.clock,
+	}
 	limiter, err := limiter.NewLimiter(
 		limiter.Config{
+			Clock: fClock,
```

Comment:
Can we change the clock in here from `timetools` to `clockwork`? 

There's a bit of dependency mess in here, so if it's too much work feel free to push back, but this may be a good opportunity to simplify.

Reply:
Yeah I think this would make a lot of sense. Since that will require changing gravitational/oxy it will need to be another PR.
Unless you just mean change it in this test and have the wrapper type just have a clockwork.clock in it instead?

Reply:
Ah, yes, now I recall. There's a pretty nasty mailgun dependency mess here.

Our limiter depends on `mailgun/ttlmap`, which depends on `mailgun/timetools`.

Note that we already _also_ depend on `gravitational/ttlmap` in other places. So the best thing would be to replace all uses of the `mailgun/ttlmap` with the gravitational one. When that's done, maybe we can get rid of `mailgun/timetools` as well.

In any case, it's out of scope for this PR. If you have time when you wrap this up to take a look I'd definitely appreciate the help and simplification!